### PR TITLE
On clicking on image now you can see the product separately

### PR DESCRIPTION
**Pull request by AbhishekDeb70**
>On clicking on the product user will be able to view it and the cart and wish-list buttons wont overlap.
>Now Admin can multiple variants for a product on the basis of size,price and color as well.
>User can view and also edit the variants accordingly

### CODE REVIEW

File: client/src/components/Admin/Products/AddForm.js [hunk 1]
```diff
@@ -10,7 +11,7 @@ const AddForm = () => {
   const navigate = useNavigate();
   const [details, setDetails] = useState({
     name: "",
-    size: "",
+    size: [],
```
Comment (iMagesh) [inline client/src/components/Admin/Products/AddForm.js:14]: what is this for? i don't see any other changes related to this? @AbhishekDeb70

Comment (iMagesh) [inline client/src/components/Admin/Products/AddForm.js:14]: what is this for? i don't see any other changes related to this? @AbhishekDeb70

**Comment by iMagesh** [comment]
@AbhishekDeb70 can you resolve the conflicts I will merge this next

**Comment by AbhishekDeb70** [comment]
> @AbhishekDeb70 can you resolve the conflicts I will merge this next

I resolved it its free to merge

File: client/src/components/Admin/Products/AddForm.js [hunk 2]
```diff
@@ -10,65 +10,67 @@ const AddForm = () => {
...
+    // ) {
+    //   setErrormsg("Price cannot be empty");
+    //   return;
+    // } else if (
+    //   !Object.keys(details).includes("description") ||
+    //   details.description === ""
+    // ) {
+    //   setErrormsg("Description cannot be empty");
+    //   return;
+    // }
```
Comment (iMagesh) [inline client/src/components/Admin/Products/AddForm.js:65]: remove all these commented code

File: server/routes/admin/products/addProducts.js [hunk 3]
```diff
@@ -18,51 +18,54 @@ router.get("/", (req, res) => {
...
+      //         console.log("congratulations u made it");
+      //       })
+      //       .catch((err) => {
+      //         res.status(400).send("Unable to post image");
+      //       });
+    //     })
+    //     .catch((err) => {
+    //       res.status(400).send("Unable to Post data ");
+    //     });
+     })
```
Comment (iMagesh) [inline server/routes/admin/products/addProducts.js:53]: remove all these commented code

File: server/routes/admin/products/addVariants.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+const fileStorageEngine = multer.diskStorage({
+  destination: (req, file, cb) => {
+    cb(null, "./images"); //"/images" is  folder storage
+  },
+  filename: (req, file, cb) => {
+    cb(null, Date.now() + "--" + file.originalname); //file.originalname has accesss to the file type
+  },
+});
+const upload = multer({ storage: fileStorageEngine });
```
Comment (iMagesh) [inline server/routes/admin/products/addVariants.js:16]: can we move this code to multer.js and place it inside utils folder?

File: server/routes/admin/products/addVariants.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  console.log("ss", req.body);
+  insertVariant(req.body)
+    .returning("variants.id")
+    .then((row) => {
+      console.log("vv", row);
+      knex("variant_images")
+        .insert({ image_url: req.file.path, variant_id: row[0].id })
+        .then((row) => {
+          res.json(row);
+          console.log("congratulations u made it");
```
Comment (iMagesh) [inline server/routes/admin/products/addVariants.js:28]: remove all the console logs and comments before committing code

File: client/src/components/Admin/Products/EditVariant.js [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+      .then((res) => {
+        setSize(res.data.size);
+        setColor(res.data.color);
+        setType(res.data.type);
+        setPrice(res.data.price);
+        setVariantId(res.data.id);
+        setProductId(res.data.pid);
+        setFileData(res.data.image_url);
+        setVariant(res.data);
+        setImageCheck(true);
```
Comment (iMagesh) [inline client/src/components/Admin/Products/EditVariant.js:60]: can we simplify this? by adding them all in a single setState?

File: client/src/components/Admin/Products/AddVariants.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+                      onChange={(e) =>
+                        setDetails({ ...details, size: e.target.value })
+                      }
+                    >
+                      <option value="0">Select product size</option>
+                      <option value="XS">XS</option>
+                      <option value="S">S</option>
+                      <option value="M">M</option>
+                      <option value="L">L</option>
+                      <option value="XL">XL</option>
```
Comment (iMagesh) [inline client/src/components/Admin/Products/AddVariants.js:74]: can you put this in an array and loop it instead of hard coding like this?

**Pull request merged**